module/lambda/main.tf

```diff
@@ -0,0 +1,29 @@
+resource "aws_lambda_function" "this" {
+  for_each = var.functions
```

Comment:
como mencione en un comment anterior, el for each adentro del modulo debe evitarse lo mas posible. El for_each deberia estar en la llamada al modulo